src/mlpack/core/cv/simple_cv_impl.hpp

```diff
@@ -0,0 +1,225 @@
+
+  return *modelPtr;
+}
+
+template<typename MLAlgorithm,
+         typename Metric,
+         typename MatType,
+         typename PredictionsType,
+         typename WeightsType>
+template<typename DataArgsTupleT,
+         typename>
+void SimpleCV<MLAlgorithm,
+              Metric,
+              MatType,
+              PredictionsType,
+              WeightsType>::Init(const float validationSize,
+                                 const DataArgsTupleT& dataArgsTuple)
+{
+  const MatType& xs = std::get<0>(dataArgsTuple);
+  const PredictionsType& ys = std::get<1>(dataArgsTuple);
+
+  Base::AssertDataConsistency(xs, ys);
+
+  size_t numberOfTrainingPoints = CalculateAndAssertNumberOfTrainingPoints(
+      validationSize, xs.n_cols);
+
+  InitTrainingAndValidationSets(xs, ys, numberOfTrainingPoints);
+}
+
+template<typename MLAlgorithm,
+         typename Metric,
+         typename MatType,
+         typename PredictionsType,
+         typename WeightsType>
+template<typename DataArgsTupleT,
+         typename,
+         typename>
+void SimpleCV<MLAlgorithm,
+              Metric,
+              MatType,
+              PredictionsType,
+              WeightsType>::Init(const float validationSize,
+                                 const DataArgsTupleT& dataArgsTuple)
+{
+  const MatType& xs = std::get<0>(dataArgsTuple);
+  const PredictionsType& ys = std::get<1>(dataArgsTuple);
+  const WeightsType& weights = std::get<2>(dataArgsTuple);
+
+  Base::AssertDataConsistency(xs, ys, weights);
+
+  size_t numberOfTrainingPoints = CalculateAndAssertNumberOfTrainingPoints(
+      validationSize, xs.n_cols);
+
+  InitTrainingAndValidationSets(xs, ys, numberOfTrainingPoints);
+
+  trainingWeights = weights.cols(0, numberOfTrainingPoints - 1);
+}
+
+template<typename MLAlgorithm,
+         typename Metric,
+         typename MatType,
+         typename PredictionsType,
+         typename WeightsType>
+size_t SimpleCV<MLAlgorithm,
+                Metric,
+                MatType,
+                PredictionsType,
+                WeightsType>::CalculateAndAssertNumberOfTrainingPoints(
+    const float validationSize,
+    const size_t total)
+{
+  if (validationSize < 0.0F || validationSize > 1.0F)
+  {
+    std::ostringstream oss;
+    oss << "SimpleCV: the validationSize parameter should be "
+        << "more than 0 and less than 1" << std::endl;
+    throw std::invalid_argument(oss.str());
+  }
+
+  if (total < 2)
+  {
+    std::ostringstream oss;
+    oss << "SimpleCV: 2 or more data points are expected" << std::endl;
+    throw std::invalid_argument(oss.str());
+  }
+
+  size_t trainingPoints = roundf(total * (1.0F - validationSize));
+
+  if (trainingPoints == 0 || trainingPoints == total)
+  {
+    std::ostringstream oss;
+    oss << "SimpleCV: the validationSize parameter is either too small "
+        << "or too big" << std::endl;
+    throw std::invalid_argument(oss.str());
```

Comment:
Yes, sure. I guess it stayed here since I used some other checking as a template.